### ISSUE TITLE
Cloning repo in named volume fails with docker rootless

### ISSUE THREAD

**Issue by ljnsn** [microsoft/vscode-remote-release#8127]
Type: <b>Bug</b>

In the bootstrap container that's created by VS Code to spin up the dev container, `/var/run/docker.sock` is mounted as a volume. When using rootless docker, that socket does not exist on the host, so a **directory** is created on the container at `/var/run/docker.sock`. Therefore, all docker commands fail and the dev container cannot be created.

When using docker rootless, instead of `/var/run/docker.sock` the correct docker socket location should be mounted, which can be read from the environment variable `DOCKER_HOST`.

Extension version: 0.282.0
VS Code version: Code 1.76.0 (92da9481c0904c6adfe372c12da3b7748d74bdcb, 2023-03-01T10:25:16.105Z)
OS version: Linux x64 6.1.11-76060111-generic
Modes:
Sandboxed: No
Remote OS version: Linux x64 6.1.11-76060111-generic

**Comment by amuntean-godaddy** [comment]
The workaround is not working when connecting to a remote rootless docker over tcp.
Is there any plan to fix it ?

**Comment by brabster** [comment]
Workaround didn't work for me. Started VSCode with DOCKER_HOST as described, cloned repo into new volume and failed with could not connect to the Docker daemon at unix:///var/run/docker.sock :sob:

**Comment by deltaclock** [comment]
> Workaround didn't work for me with local docker rootless either. Started VSCode with DOCKER_HOST as described, cloned repo into new volume and failed with could not connect to the Docker daemon at unix:///var/run/docker.sock 😭

Hi, usually the _rootless_ docker socket is on another path, for example `/run/user/1000` on my system. Try searching for that instead.

**Comment by brabster** [comment]
@deltaclock yup, I have the env var set to the correct socket (also uid 1000 like you) but vscode devcontainers still seems to be looking for `/var/run/docker.sock`, guessing something's hardcoded instead of using DOCKER_HOST
